Robot Code/src/org/usfirst/frc/team3316/robot/commands/chassis/autonomous/DriveDistance.java

```diff
@@ -46,96 +45,79 @@ public void setPIDSourceType(PIDSourceType pidSource)
 
 			public double pidGet()
 			{
-				return Robot.chassis.getRightSpeed();
-			}
-
-			public PIDSourceType getPIDSourceType()
-			{
-				return PIDSourceType.kRate;
-			}
-		}, new PIDOutput()
-		{
-
-			public void pidWrite(double output)
-			{
-				pidRightOutput = output;
-			}
-		});
+				currentDist = Robot.chassis.getDistance() - initDist;
 
-		pidLeft = new PIDController(0, 0, 0, new PIDSource()
-		{
+				SmartDashboard.putNumber("Current Distance", currentDist);
 
-			public void setPIDSourceType(PIDSourceType pidSource)
-			{
-				return;
-			}
+				if (currentDist > dist + (double) config
+						.get("chassis_DriveDistance_PID_Tolerance")
+						&& currentDist < dist - (double) config
+								.get("chassis_DriveDistance_PID_Tolerance"))
+				{
+					pid.reset();
```

Comment:
I heard that we're not using an integral term anyway. Why then do we need to reset the pid controller?